R/getDominantTaxa.R

```diff
@@ -0,0 +1,152 @@
+#'
+#' @return \code{x} with additional \code{\link{colData}} named
+#'   \code{*name*}
+#'
+#' @name getDominantTaxa
+#' @export
+#'
+#' @author Leo Lahti and Tuomas Borman. Contact: \url{microbiome.github.io}
+#'
+#' @examples
+#' data(GlobalPatterns)
+#' x <- GlobalPatterns
+#'
+#' #Finds the dominant taxa.
+#' x <- getDominantTaxa(x)
+#' #Information is stored to colData
+#' colData(x)
+#'
+#' #If taxonomic information is available, it is possible to find the most dominant
+#' #group from specific taxonomic level, here family level. The name of column can be specified.
+#' x <- getDominantTaxa(x, rank="Family", name="Dominant Family")
+#' colData(x)
+#'
+#' x <- microbiomeDataSets::dietswap()
+#' #With group, it is possible to group observations based on group specified by user
+#' x <- getDominantTaxa(x, group = "nationality")
+#'
+NULL
+
+#' @rdname getDominantTaxa
+#' @export
+setGeneric("getDominantTaxa",signature = c("x"),
+           function(x, rank = NULL, group = NULL, name = "dominant_taxa")
+               standardGeneric("getDominantTaxa"))
+
+
+#' @rdname getDominantTaxa
+#' @export
+setMethod("getDominantTaxa", signature = c(x = "SummarizedExperiment"),
+          function(x, rank = NULL, group = NULL, name = "dominant_taxa"){
+
+              #Input check
+              #rank check
+              if(!is.null(rank)){
+                  if(!.is_a_string(rank)){
+                      stop("'rank' must be an single character value.",
+                           call. = FALSE)
+                  }
+                  .check_taxonomic_rank(rank, x)
+              }
+
+              #group check
+              if(!is.null(group)){
+                  if(isFALSE(any(group %in% colnames(colData(x))))){
+                      stop("'group' variable must be in colnames(colData(x))")
+                  }
+              }
+
+              #name check
+              if(!.is_non_empty_string(name)){
+                  stop("'name' must be a non-empty single character value.",
+                       call. = FALSE)
+              }
+
+              #If "rank" is not NULL, species are aggregated according to the taxonomic rank
+              #that is specified by user.
+              if (!is.null(rank)) {
+                  x <- agglomerateByRank(x, rank = rank)
+              }
+
+              #assays()$counts returns counts of the taxa in the samples
+              #apply() function finds the indices of taxa's that has the highest
+              #amount of counts.
+              #names() returns the names of taxa that are the most abundant.
+              taxas <- names(x)[apply(assays(x)$counts, 2, which.max)]
+
+              #Adds taxas to colData
+              x <- .add_dominant_taxas_to_colData(x, taxas, name)
+
+              #Gets an overview
+              overview <- .get_overview(x, group, name)
+
+              #Prints the overview
+              print(overview)
```

Comment:
Never use `print`, if the user doesn't specifically asks for it, like in `show`.

Use `message` instead, since this can be turned off by using `suppressMessages` at the users conveniance

Reply:
Changed to message()